Drag drop group name

VacuumBreather opened the pull request:
## Motivation / Problem

Naming groups is a nuisance and takes forever. A little assistance by the GUI would be nice.

## Description

This feature allows the generation of a group with a pre-generated name by dragging and dropping a vehicle onto the new group button. If Ctrl is held while doing this all vehicles with the same orders list will be added as well.

There is probably no way to create a group generation scheme that makes everybody happy. But loaded cargo and origin and destination are usually the standard. This generates a baseline name that players can modify. But I think we can agree that everybody needs the town names in there somehow and very often the cargo. Why type all that if it can be generated. Better modify a name than type it all every time. And for a small game these group names are sufficient by themselves.

I borrowed some code from the auto-generate-groups PR. So if this is merged before the problems with that one are resolved, that code can be reused and is already in place.

## Limitations

The cargo list might be a bit long if multiple cargos are loaded or supply waggons are attached like some people do when playing FIRS. So maybe a setting could be added to deactivate the adding of the cargo list.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

JGRennison commented:
Using this on a vehicle with no orders causes a crash.

Special orders type (e.g. conditional orders) cause random stations/towns to be selected.
Probably only OT_GOTO_STATION should be used instead of only discarding OT_GOTO_DEPOT.

Doing name generation inside the command makes localisation and setting-controlled generation a bit more problematic. That's not a big priority for now though.

Doing create group from the list form the station, waypoint and depot vehicle lists acts like the shared order mode for one of the windows, but in that case an auto-generated name is probably not appropriate anyway.

I'll have a go at these later on the week.

VacuumBreather commented:
Crash is fixed. I always forget that you cannot have additional error checks after you go into the EXEC branch.
Anyway, I also added an error message now in case the group creation fails.

I reverted the earlier change so the other two methods of creating a group are back to the original. I guess if you use those you have other more specific groups in mind so they probably shouldn't work like this after all.

I thought about maybe checking if every vehicle in the group has the same orders and every vehicle with those orders is in the list, and trigger the command under those circumstances. But that seemed unintuitive to the user and unexpected. If you use a list window for group creation you probably want a very specific group. I think this should be limited to the drag and drop and possibly later extended to a mass-auto-group feature if anybody wants to tackle that.

The error with the orders was fixed and that should probably be done in the other PR as well.
